File: iteration2/Models/Lab.java
package iteration2.Models;

import org.json.JSONObject;

import java.util.ArrayList;

public class Lab extends Course {
    private int sectionNumber;

    public Lab(){}
    public Lab(String name, String code, int credit, int requiredCredits, int quota, int semester,
               ArrayList<Course> preRequisiteCourses, ArrayList<String> weeklyHours) {
        this.setName(name);
        this.setCode(code);
        this.setCredit(credit);
        this.setRequiredCredits(requiredCredits);
        this.setQuota(quota);
        this.setSemester(semester);
        this.setPreRequisiteCourses(preRequisiteCourses);
        this.setWeeklyHours(weeklyHours);
        this.setSectionNumber(generateSectionNumber());
    }

    public int generateSectionNumber(){
        return (int)Math.floor(Math.random()*(4)+1);
    }
    public int getSectionNumber() {
        return sectionNumber;
    }

    public void setSectionNumber(int sectionNumber) {
        this.sectionNumber = sectionNumber;
    }

    @Override
    public JSONObject toJson() {
        return null;
    }
}

File: iteration3/Models/Lab.py
import json
import math
import random
from typing import List

from iteration3.Models.Course import Course


class Lab(Course):
    def __init__(self, name, code, credit, required_credits, quota, semester,
                 pre_requisite_courses, weekly_hours):
        super().__init__(name, code, credit, required_credits, quota, semester, pre_requisite_courses, weekly_hours)
        self.__section_number = self._generate_section_number()

    def _generate_section_number(self):
        return math.floor(random.random() * 4 + 1)

    def get_credit(self):
        return self.__credit

    def set_section_number(self, section_number: int):
        self.section_number = section_number

    def to_json(self) -> json:
        return None
